doku-derive/src/expand.rs

```diff
@@ -34,3 +25,22 @@ pub fn expand(input: &syn::DeriveInput) -> Result<TokenStream2> {
         .into()),
     }
 }
+
+fn new_generics_with_where_clause(
+    generics: &syn::Generics,
+) -> Result<syn::Generics> {
+    let mut new_generics = generics.to_owned();
+    let where_clause = new_generics.make_where_clause();
+    for generic in &generics.params {
+        match generic {
+            syn::GenericParam::Const(_) => (),
+            syn::GenericParam::Lifetime(_) => (),
+            syn::GenericParam::Type(t) => {
+                let predicate: syn::WherePredicate =
+                    syn::parse2(quote! { #t: ::doku::Document }).unwrap();
+                where_clause.predicates.push(predicate);
```

Comment:
Pretty cool approach, I've had no idea adding support for generics is that easy - I've always thought it'll take like a hundred lines of code or so 😄